readme.md

```diff
@@ -81,4 +81,30 @@ Clone nivio, build and run a Docker image:
   
   Open http://localhost:8080
   
-  If you want to contribute to our frontend, read further into our [Frontend Readme](https://github.com/dedica-team/nivio/tree/develop/src/main/app)
\ No newline at end of file
+  If you want to contribute to our frontend, read further into our [Frontend Readme](https://github.com/dedica-team/nivio/tree/develop/src/main/app)
+  
+**Nivio Backend Architecture**
+
+If you want to contribute to our backend, maybe the following diagram is of use to you. It shows some of the most important classes and 
+interfaces. It is supposed to give you an idea on how the backend is structured, but note that not all details are displayed:
+
+ ![layoutedArtifact graph](backend_architecture_api.png)
+ 
+ If you use the `ApiController` as the entry point to the backend, you can see that it retrieves information and triggers events
+  to the most important parts of the application. 
+ 
+The `LandscapeRespository` gives access to the stored landscapes.
+  
+The `LandscapeDescriptionFactory` is used to generate a `LandscapeDescription` from various sources, such as a `String` input, 
+or e.g. from a yaml file. 
+
+This `LandscapeDescription` has to be enriched with the `ItemDescription` for all items in the landscape.
+This is managed by the `InputFormatHandler`, which are able to read several input formats such as e.g. kubernetes files, or the
+nivio description format. Access to these handlers is managed by the `InputFormatHandlerFactory`.
+
+To actually create a landscape, the `Indexer` is used. This is able to compute the landscape graph from a `LandscapeDescription`.
+The `Indexer` uses several `Resolver` to resolve groups and item relations in landscapes and e.g. the appearance of the graph.
+
+The `Indexer` can be triggered either directly through the `ApiController` to index or reindex a landscape, or it is triggered
+by an observer mechanism on files. These are the files located under the path provided through the `SEED` environment variable.
```

Comment:
@bonndan : we discussed above that the `Indexer` is triggered only by an `ApplicationEvent`. In the `ApiController` though, the endpoints `indexLandscape()` and `reindex()` both also trigger the `Indexer`.

Is my description correct?

Reply:
we should maybe conside limiting the triggering to events, this way the frontend is notified of the event, too.

Reply:
I'll open a new ticket.